machine-check/src/swarm.rs

```diff
@@ -175,6 +183,38 @@ fn to_swarm(graph: &Graph) -> super::Graph {
     graph.map(|_, n| n.name.clone(), |_, x| x.clone())
 }
 
+fn all_nodes_reachable(graph: &Graph, initial: NodeId) -> Vec<Error> {
```

Comment:
You can make this simpler by creating two sets:
- `A` - One with all nodes
- `B` - Another will reachable/visited nodes from the starting ones

`A - B` will give you the unreachable nodes, the code becomes simpler and possibly more performant as well.

Extra note: There's an extra check we might want to add (in another PR) which is for states without outgoing edges, but this one needs the concept of final states in the machine runner to work.

Reply:
Good idea. Though, sticking with vec for B since &A and &B constructs a new BTreeSet

Reply:
You just need to list the unreachable ones, order isn't an issue so I don't see the problem with the BTreeSet. What am I missing?

Reply:
No, yes I'm applying your A - B in terms of operations of abstract list (see my latest commit).

But I'm not subscribing to the A - B if the idea you're referring to is that std::collections::BTreeSet implements std::ops::Sub.

If BTreeSub implements std::ops::SubAssign I would use it

Reply:
Ah, the `A-B` isn't Rust, just the difference from set theory
![image](https://github.com/Actyx/machines/assets/15343819/fd04768b-4f1c-448f-ad87-4f74bcb70476)